jsdoc/services/code-name-matchers/export-named-declaration.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * @dgService ExportNamedDeclarationNodeMatcher
+ * @returns {String|Null} code name from node
+ */
+module.exports = function ExportNamedDeclarationNodeMatcherFactory (codeNameService) {
+  /**
+   * @param {Node} node AST node to process
+   * @returns {String|Null} code name from node
+   */
+  return function ExportNamedDeclarationNodeMatcher (node) {
+    return null;
```

Comment:
Shouldn't this be returning the name of the export here?

Reply:
I figured it was supposed to work as https://github.com/angular/dgeni-packages/blob/master/jsdoc/services/code-name-matchers/export-default-declaration.js
But then is should return `return node.id && node.id.name || null;`instead?

Reply:
If I understand correctly, default exports have no name, so we return `null` but by definition named exports have a name. We just need to work out what property on the AST node will give us that name.

Reply:
Perhaps @wingedfox has suggestions since he originally coded this feature.

Reply:
I'm pretty sure default exports can (and usually) have names. For example:
```javascript
export default class Car {
    constructor() {}
}
```
Maybe this is a limitation in the AST implementation?

Reply:
Good point, perhaps an oversight on our part?

Reply:
AST describes the tree only and does not force a node to have properties. `id` property has a meaning for only a small list of nodes which may be referenced (evaluated) from the execution context.
Perhaps, for default export we have to run matcher against a right subtree to find out which node is down there. As long as ES allows to export virtually anything you can't rely on a fixed structure of the right subtree.
Take a look at https://github.com/angular/dgeni-packages/blob/master/jsdoc/services/code-name-matchers/assignment-expression.js

Reply:
Is there anywhere i can try and generate AST to see what it looks like?

Reply:
https://astexplorer.net/?

Reply:
https://astexplorer.net/